runtime/script_loader.cpp

```diff
@@ -166,15 +272,59 @@ JSObject* module_resolve_hook(JSContext* cx, HandleValue referencingPrivate,
   return get_module(cx, path.get(), resolved_path, opts);
 }
 
- ScriptLoader::ScriptLoader(JSContext *cx, JS::CompileOptions *opts) {
+bool module_metadata_hook(JSContext* cx, HandleValue referencingPrivate, HandleObject metaObject) {
```

Comment:
This hook sets the builtin as `import.meta.builtin` when the module is a builtin module.